### DISCUSSION TITLE
Ad impressions、cost are negative，why？

### DISCUSSION THREAD

**Discussion by Leibosite** [amzn/ads-advanced-tools-docs#248] [Q&A] [answered]
Ad impressions are negative, how to deal with these data? Is this data illegal?

**Comment by elizafarley** [comment]
@Leibosite is this something you're seeing in the reporting API or Amazon Marketing Stream? In the reporting API, this would likely be a bug. Amazon Marketing Stream will send negative values, since there are sometimes restatements to the data.

**Reply by Leibosite** [reply]
Thank you, I am being troubled！Do negative values have meaning? How to use these negative values? Should these negative values be filtered out during normal statistics?

**Comment by kdwivedi13** [comment]
@Leibosite 
Thanks for bringing this to our attention. It seems we need to investigate this further. To assist us, could you please create a [support ticket ](https://amzn-clicks.atlassian.net/servicedesk/customer/portal/2/group/2)with more details? Specifically, please include the request payload, timeframe of the issue, and whether it's related to a stream. Additionally, if it is a stream issue, please provide details about the dataset and advertiser involved.

**Reply by Leibosite** [reply]
yes，support ticket is a good idea，But I am a developer and cannot log into the Amazon advertiser account, so I cannot support the ticket.In the AWS data set I subscribe to, there are a lot of negative values for impressions, clicks, and cost.This should be a general logic, I'm very curious about the meaning of negative values,This should be a common logic. I am very curious about the meaning of negative values. Can you contact me by email? leibosite@gmail.com

**Reply by elizafarley** [reply, marked as answer]
In general, some negatives are expected due to restatements in the data, but they should always aggregate to positive values. 

See this FAQ for more info: https://advertising.amazon.com/API/docs/en-us/guides/amazon-marketing-stream/onboarding#why-am-i-seeing-negative-values-for-the-first-couple-of-days-of-data.

You can also look at the sample data and query here to see an example of negative data coming through: https://advertising.amazon.com/API/docs/en-us/guides/amazon-marketing-stream/overview#sample-query

**Reply by Leibosite** [reply]
Thank you very much. This solved a lot of my doubts. You are very professional. Thank you again.

**Comment by Leibosite** [comment]
In general, some negatives are expected due to restatements in the data, but they should always aggregate to positive values.

See this FAQ for more info: https://advertising.amazon.com/API/docs/en-us/guides/amazon-marketing-stream/onboarding#why-am-i-seeing-negative-values-for-the-first-couple-of-days-of-data.

You can also look at the sample data and query here to see an example of negative data coming through: https://advertising.amazon.com/API/docs/en-us/guides/amazon-marketing-stream/overview#sample-query
